python/ray/experimental/sgd/pytorch/pytorch_runner.py

```diff
@@ -0,0 +1,173 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import logging
+import os
+import torch
+import torch.distributed as dist
+import torch.utils.data
+
+import ray
+
+from ray.experimental.sgd.pytorch import utils
+
+logger = logging.getLogger(__name__)
+
+
+class PyTorchRunner(object):
+    """Manages a distributed PyTorch model replica"""
+
+    def __init__(self,
+                 model_creator,
+                 data_creator,
+                 optimizer_creator,
+                 config=None,
+                 batch_size=16,
+                 backend="gloo"):
```

Comment:
let's actually default to `nccl`.

Reply:
[`nccl` only works with GPU](https://pytorch.org/docs/stable/distributed.html) so I think it would be better to default to `gloo` or `mpi`.

Reply:
Hm, I'd do an `auto` that defaults to gloo and flips to nccl when on GPU. The performance is so much better with NCCL on a properly configured GPU, and when users switch onto GPU from CPU, this is option is not clearly surfaced..